src/cobalt.rs

```diff
@@ -120,6 +124,57 @@ pub fn build(config: &Config) -> Result<()> {
         }
     }
 
+    // load data files
+    let mut data_map : HashMap<String, Value> = HashMap::new();
+    let data_root = source.join("_data");
+    let data_files_builder = FilesBuilder::new(data_root.as_path())?;
+    let data_files = data_files_builder.build()?;
+
+    for df in data_files.files() {
+
+        if let Some(ext) = df.extension() {
+            if let Some(file_stem) = df.file_stem() {
+                let file_name = String::from(file_stem.to_str().unwrap());
+                let full_path = data_root.join(df.clone());
+                let data : Value;
+
+                if ext == OsStr::new("yml") || ext == OsStr::new("yaml")  {
+                    let reader = File::open(full_path)?;
+                    data = serde_yaml::from_reader(reader)?;
+                } else if ext == OsStr::new("json") {
+                    let reader = File::open(full_path)?;
+                    data = serde_json::from_reader(reader).unwrap();
```

Comment:
Though the serde_json examples show they are using `?`, if I replace the unwrap there, I get:
```
error[E0277]: the trait bound `error::Error: std::convert::From<serde_json::Error>` is not satisfied
   --> src/cobalt.rs:146:28
    |
146 |                     data = serde_json::from_reader(reader)?;
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `std::convert::From<serde_json::Error>` is not implemented for `error::Error`

```

Any ideas or hints why that might be?

Same for the toml version

Reply:
I believe that means we need to chain our error type to their error types so that error-chain will implement `From` for us.

See https://github.com/cobalt-org/cobalt.rs/blob/master/src/error.rs

Speaking of, I recently added some `chain_err` to our code to call out what file a process failed on.  I'd recommend applying it here.

See https://github.com/cobalt-org/cobalt.rs/commit/1e07708badd73523892e4ab7a7c17c40d090bae2#diff-1adad72e4e1dbe7019b775a6acb9455aR182